scanrepository/scanrepository.go

```diff
@@ -515,9 +515,32 @@ func (cfp *ScanRepositoryCmd) getOpenPullRequestBySourceBranch(branchName string
 func (cfp *ScanRepositoryCmd) aggregateFixAndOpenPullRequest(vulnerabilitiesMap map[string]map[string]*utils.VulnerabilityDetails, aggregatedFixBranchName string, existingPullRequestInfo *vcsclient.PullRequestInfo) (err error) {
 	log.Info("-----------------------------------------------------------------")
 	log.Info("Starting aggregated dependencies fix")
+
+	workTreeIsClean, err := cfp.gitManager.IsClean()
+	var tempDirPath string
+	if !workTreeIsClean {
+		tempDirPath, err = utils.CopyCurrentDirFilesToTempDir()
+		if err != nil {
+			return
+		}
+		defer func() {
+			err = errors.Join(err, fileutils.RemoveTempDir(tempDirPath))
+		}()
+	}
+
 	if err = cfp.gitManager.CreateBranchAndCheckout(aggregatedFixBranchName); err != nil {
 		return
 	}
+
+	if !workTreeIsClean {
+		// We copy to the new branch all files the working directory contained before changing branch
+		// We perform this so we will not have to 'install' the project again if it was already done before
+		err = utils.CopyMissingFilesToCurrentWorkingDir(tempDirPath)
+		if err != nil {
+			return
+		}
+	}
+
```

Comment:
Let's move these logic into a new function and call it from both aggregate and non-aggregate functions, if possible call it once before calling these function. Also, document the new function well.